src/common/dao/projectmember.go

```diff
@@ -15,66 +15,80 @@
 package dao
 
 import (
+	"github.com/vmware/harbor/src/common"
 	"github.com/vmware/harbor/src/common/models"
 )
 
 // AddProjectMember inserts a record to table project_member
-func AddProjectMember(projectID int64, userID int, role int) error {
+func AddProjectMember(projectID int64, userID int, role int, entityType string) (int, error) {
 	o := GetOrmer()
+	if !(entityType == common.UserMember || entityType == common.GroupMember) {
+		entityType = common.UserMember
+	}
+	sql := `insert into project_member (project_id, entity_id , role, entity_type) values (?, ?, ?, ?)`
+	_, err := o.Raw(sql, projectID, userID, role, entityType).Exec()
+	if err != nil {
+		return 0, err
+	}
+	var pmid int
+	querySQL := `select id from project_member where project_id = ? and entity_id = ? and entity_type = ? limit 1`
 
-	sql := "insert into project_member (project_id, user_id , role) values (?, ?, ?)"
-
-	_, err := o.Raw(sql, projectID, userID, role).Exec()
-
-	return err
+	err = o.Raw(querySQL, projectID, userID, entityType).QueryRow(&pmid)
+	if err != nil {
+		return 0, err
+	}
+	return pmid, err
 }
 
 // UpdateProjectMember updates the record in table project_member
-func UpdateProjectMember(projectID int64, userID int, role int) error {
+func UpdateProjectMember(projectID int64, userID int, role int, entityType string) error {
```

Comment:
rename the parameter "userID"  and I don't think we need entityType in the param list.